shelley/design-spec/delegation_design_spec.tex

```diff
@@ -4468,6 +4470,162 @@ \subsection{Won't stake pools reject delegation certificates that delegate away
 partake in malicious behaviour and attack the system, against their direct
 incentives.
 
+\section{Transaction Metadata}
+
+Adding metadata to transactions is a useful new feature in Cardano Shelley.
+It is not related to delegation or decentralisation.
+
+\subsection{Motivation and design goals}
+
+The purpose is to enable a range of new applications by allowing arbitrary
+structured data to be included onto the chain, and to make effective use of
+that data. The term `metadata' is perhaps a misnomer since it is simply about
+placing application specific data on the chain; it is only metadata from the
+point of view of a transaction since it is carried along with transactions and
+not involved in validation.
+
+A design goal is to add very little complexity to the on-chain part of the
+system but to get (or allow for) as much functionality as possible, in
+combination with other features or components. This helps keep implementation
+complexity lower. Importantly it keeps the size of the trusted base low, by
+having the complex functionality to use the metadata outside of the trusted
+base.
+
+A design principle that we preserve is that the historical data on the chain is
+not needed to validate the next block or transaction. All data needed for later
+validation must be explicitly tracked in the ledger state. This means the old
+part of the chain does not need to be preserved locally at all, or at least not
+in random access storage. This avoids a problem that Ethereum ran into with
+disk I/O becoming a performance bottleneck. This is why the design does not
+include metadata into the ledger state, and does not make it accessible to
+later scripts.
+
+\subsection{Detail}
+
+The transaction can contain metadata. The metadata hash is part of the body of
+the transaction so is covered by all transaction signatures. The metadata value
+is kept with the transaction witnesses. This follows the `segmented witness'
```

Comment:
I'm not sure if "with the transaction witnesses" is potentially confusing. it is indeed outside the body in segwit style, but is separate from the witnesses. hence the "merkle triple".

Reply:
Yeah, I wasn't sure how much detail to go into here. Perhaps I can adjust that to be accurate without being too precise.

Reply:
Pushed an update.